pkg/auth/local_authenticator.go

```diff
@@ -0,0 +1,113 @@
+package auth
+
+import (
+	"crypto"
+	"net/http"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/go-openapi/runtime"
+	"github.com/go-openapi/runtime/security"
+	"github.com/go-openapi/strfmt"
+	"github.com/jinzhu/gorm"
+	"github.com/openshift/assisted-service/internal/cluster"
+	"github.com/openshift/assisted-service/internal/common"
+	logutil "github.com/openshift/assisted-service/pkg/log"
+	"github.com/openshift/assisted-service/pkg/ocm"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type LocalAuthenticator struct {
+	log       logrus.FieldLogger
+	publicKey crypto.PublicKey
+	db        *gorm.DB
+}
+
+func NewLocalAuthenticator(cfg *Config, log logrus.FieldLogger, db *gorm.DB) (*LocalAuthenticator, error) {
+	if cfg.ECPublicKeyPEM == "" {
+		return nil, errors.Errorf("local authentication requires an ecdsa Public Key")
+	}
+
+	key, err := jwt.ParseECPublicKeyFromPEM([]byte(cfg.ECPublicKeyPEM))
+	if err != nil {
+		return nil, err
+	}
+
+	a := &LocalAuthenticator{
+		log:       log,
+		publicKey: key,
+		db:        db,
+	}
+
+	return a, nil
+}
+
```

Comment:
Since LocalAuthenticator must match the Authenticator interface, I think we should add a compile-time assert on it
`var _ Authenticator = (*LocalAuthenticator){}`

Reply:
Done